tests/state/test_state_api.py

```diff
@@ -63,3 +64,19 @@ def test_get_states_by_name(self, n, stub_broker, state_middleware, api_client):
         # check integrity
         for state in states:
             assert state in random_states
+
+    def test_request_cancel(self, stub_broker, api_client, state_middleware, cancel_backend):
+        message_id = "3141516"
+        state = State(message_id, StateNamesEnum.Pending, [], {})
+        state_middleware.backend.set_state(state, ttl=1000)
+        stub_broker.add_middleware(Cancel(backend=cancel_backend))
+        api_client.post("/messages/cancel/{}".format(message_id))
+        assert cancel_backend.is_canceled(message_id, None)
+
+    def test_request_cancel_without_backend(self, stub_broker, api_client, state_middleware):
+        message_id = "3141516"
+        state = State(message_id, StateNamesEnum.Pending, [], {})
+        state_middleware.backend.set_state(state, ttl=1000)
+        res = api_client.post("/messages/cancel/{}".format(message_id))
+        # if there is not cancel backend should raise an error
+        assert res.status_code == 500
```

Comment:
could you add a check on the error message : `assert res.json = NoResultBackend` or something like this.